File: WebContent/org/r3/db/system/siteportletcfg/SitePortletConfigDTO.js
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//



jQuery.r3Utils.namespace("org.r3.db.system.siteportletcfg");

org.r3.db.system.siteportletcfg.SitePortletConfigDTO = Class.extend(
{
	 init: function(aDataBean)
	 {  
	 	 	 this.dtrowclass   = "grade_siteportletconfig";
       this.dtokey       = aDataBean.dtokey   || null;
       this.dtodescr     = aDataBean.dtodescr || null;
       
        this.site = aDataBean.site || null;
        this.pageid = aDataBean.pageid || null;
        this.positionid = aDataBean.positionid || null;
        this.jsclassname = aDataBean.jsclassname || null;
        this.cssclassname = aDataBean.cssclassname || null;
        this.datacardid = aDataBean.datacardid || null;
        this.dataoptions = aDataBean.dataoptions || null;
        this.dataonuserstatus = aDataBean.dataonuserstatus || null;
 
	 }
    
  ,compareTo: function(anItemId)
   {
   	   if (!this.dtokey)
   	       return -99;
   	       
   	   if (this.dtokey < anItemId)
   	       return -1;
   	   else if (this.dtokey > anItemId)
   	   	   return  1;
   	   	   
   	   return 0;
   }
   
});

File: src/org/r3/db/system/siteportletcfg/SitePortletConfigDTO.java
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//


package org.r3.db.system.siteportletcfg;


public class SitePortletConfigDTO extends SitePortletConfigDTOBase
{
	public enum PortletOnUserStatusActivation
	{
		userlogged, usernotlogged, always;
		
		public static PortletOnUserStatusActivation getValue4(String s, PortletOnUserStatusActivation defaultValue)
		{
			if (s != null && s.length() > 0)
				return PortletOnUserStatusActivation.valueOf(s);
			
			return defaultValue;
		}
	}
	
	public static SitePortletConfigDTO NullPortlet = new SitePortletConfigDTO();
	
    public SitePortletConfigDTO()
    {
        super();
    }

    public SitePortletConfigDTO(SitePortletConfigDTO another)
    {
        super(another);
    }

    public String getHTML4Position(String positionId, boolean isUserLogged)
	{
       	StringBuilder stb = new StringBuilder();
    	
    	PortletOnUserStatusActivation oua = PortletOnUserStatusActivation.getValue4(getDataonuserstatus(), PortletOnUserStatusActivation.always);
    	if (oua == PortletOnUserStatusActivation.always || 
    		 (oua == PortletOnUserStatusActivation.userlogged && isUserLogged) || 
    		 (oua == PortletOnUserStatusActivation.usernotlogged && !isUserLogged))
    	{	

   		if (getJsclassname() != null)
    	    stb.append(" class='").append(getJsclassname()).append("' ");
    	
    	if (getDatacardid() != null)
    		stb.append(" data-cardid='").append(getDatacardid()).append("' ");

    	if (getDataoptions() != null)
    		stb.append(" data-options='").append(getDataoptions()).append("' ");
    	
    	}
    	
    	return stb.toString();	    
	}
}
